src/layout/Popup.js

```diff
@@ -0,0 +1,157 @@
+    Popup.prototype.publish("width", null, "number", "Popup width",null,{});
+    Popup.prototype.publish("height", null, "number", "Popup height",null,{});
+    Popup.prototype.publish("top", null, "number", "Top position property of popup",null,{});
+    Popup.prototype.publish("bottom", null, "number", "Bottom position property of popup",null,{});
+    Popup.prototype.publish("left", null, "number", "Left position property of popup",null,{});
+    Popup.prototype.publish("right", null, "number", "Right position property of popup",null,{});
+    Popup.prototype.publish("position", "relative", "set", "Value of the 'position' property",["absolute", "relative", "fixed", "static", "initial", "inherit" ],{tags:['Private']});
+    
+    Popup.prototype.publish("buttonName", null, "string", "Button name property",null,{});
+    Popup.prototype.publish("buttonLabel", "", "string", "Button label text",null,{});
+
+    Popup.prototype.publish("widget", null, "widget", "Widget",null,{tags:['Private']});
+
+
+    Popup.prototype.testData = function () {
+        this.showTestButton(true);
+        this.popupState(false);
+        this.position("absolute");
+        this.top(30);
+        this.width(400);
+        this.height(500);
+        
+        var context = this;
+        this._testButton = new Input()
+            .type("button")
+            .value("Open Popup")
+            .on("click", function () {
+                context.updateState(!(context.popupState()));
+            }, true)
+        ; 
+
+        this.widget(new Surface().widget(new Column().testData()));
+
+        return this;
+    };
+
+    Popup.prototype.widgetSize = function (widgetDiv) {
+        var width, height;
+        
+        if (this.width()) {
+            width = this.width();
+        } else {
+            width = this.clientWidth() - this.calcFrameWidth(widgetDiv);
+        }
+        if (this.height()) {
+            height = this.height();
+        } else {
+            height = this.clientHeight() - this.calcFrameHeight(widgetDiv);
+        }
+ 
+        return { width: width, height: height };
+    };
+    
+    Popup.prototype.updateState = function (visible) {
+            this.popupState(visible).update(this._domNode, this._element);
+    };
+    
+    Popup.prototype.enter = function (domNode, element) {
+        HTMLWidget.prototype.enter.apply(this, arguments);
+
+        if (this._testButton) {
+            this._testButton.target(domNode.parentNode).render();
+        }
+        this._domNode = domNode;
+        this._element = element;
+    };
+
+    Popup.prototype.update = function (domNode, element) {
+        HTMLWidget.prototype.update.apply(this, arguments);
+        var context = this;        
+
+        element
+            .style("position",this.position())
+            .style("left",this.left() + "px")
+            .style("right",this.right() + "px")
+            .style("top",this.top() + "px")
+            .style("bottom",this.bottom() + "px")
```

Comment:
If these styles were published for the existing surface, would it "work" as a the popup surface?